File: Krootify.Angular/src/app/api/models/subscription-info.ts
/* tslint:disable */
/* eslint-disable */
import { PlanPeriod } from './plan-period';
import { SubscriptionStatus } from './subscription-status';
export interface SubscriptionInfo {
  autoRenewStatus?: boolean;
  endDate?: string;
  id?: number;
  planPeriod?: PlanPeriod;
  price?: number;
  productId?: null | string;
  startDate?: string;
  subscriptionStatus?: SubscriptionStatus;
  subscriptionId?: null | string;
  userId?: null | string;
}

File: Krootify.Domain/SubscriptionInfo.cs
using Domain.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain
{
    public class SubscriptionInfo : BaseEntity
    {
        public string SubscriptionId { get; set; }

        public PlanPeriod PlanPeriod { get; set; }

        public string UserId { get; set; }

        public DateTime ExpireAt { get; set; }

        public bool AutoRenewStatus { get; set; }

        public DateTime StartDate { get; set; }

        public decimal Price { get; set; }

        public string ProductId { get; set; }

        public SubscriptionStatus Status { get; set; }
    }
}
